### PR TITLE
#21209: Reorganized remote reference.conf and its docs

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

File: akka-remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -937,3 +651,313 @@ akka {
...
+       # It's only called from the stage, so if it dosn't delegate to any shared instance
+       # it doesn't have to be thread-safe.
+       # Refer to `akka.remote.artery.RemoteInstrument` for more information.
+       instruments = ${?akka.remote.artery.advanced.instruments} []
+      }
+    }
+  }
+
+}
+#//#artery
```
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: line break here is important (when merging of files for sbt assembly or similar)

**Review by patriknw — APPROVED** [review]
LGTM apart from a newline

File: akka-remote/src/main/resources/reference.conf [hunk 2]
```diff
@@ -937,3 +651,313 @@ akka {
...
+    artery {
+
+      # Enable the new remoting with this flag
+      enabled = off
+
+      # Canonical address is the address other clients should connect to.
+      # Artery transport will expect messages to this address.
+      canonical {
+
+       # The default remote server port clients should connect to.
```
Comment (johanandren) [inline akka-remote/src/main/resources/reference.conf]: odd indent

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
this has been validated locally and the jenkins jobs has passed remoting and some cluster tests

**Comment by patriknw** [comment]
Refs #21209

**Comment by drewhk** [comment]
and the test passed in the end ;)

**Comment by patriknw** [comment]
yeah, only 5 hours latency

**Comment by ktoso** [comment]
I hear we may get another Jenkins machine on a7. Will catch up with Toni
about it

**Comment by patriknw** [comment]
yes, perhaps we should give a2 a new chance also. It had that weird javadoc error. Perhaps we could use it as a multi-node slave at least.

**Pull request merged**